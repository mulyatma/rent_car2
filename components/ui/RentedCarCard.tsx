/* eslint-disable prettier/prettier */
import React from 'react';
import { View, Text, StyleSheet, Image } from 'react-native';

interface Car {
    carId: {
        nameCar: string;
        img: string;
    };
    startDate: string;
    endDate: string;
    totalPrice: number;
}

interface RentedCarCardProps {
    car: Car;
}

export default function RentedCarCard({ car }: RentedCarCardProps) {
    const formatDate = (date: string): string => {
        const d = new Date(date);
        return `${d.getDate()}/${d.getMonth() + 1}/${d.getFullYear()}`;
    };

    const formatPrice = (price: number): string => {
        return price.toString().replace(/\B(?=(\d{3})+(?!\d))/g, '.');
    };

    return (
        <View style={styles.card}>
            <View style={styles.textContainer}>
                <Text style={styles.cardName}>{car.carId.nameCar}</Text>
                <Text style={styles.cardDescription}>Sewa dari: {formatDate(car.startDate)}</Text>
                <Text style={styles.cardDescription}>Sampai: {formatDate(car.endDate)}</Text>
                <Text style={styles.cardPrice}>Total harga: Rp {formatPrice(car.totalPrice)}</Text>
            </View>
            <Image source={{ uri: car.carId.img }} style={styles.cardImage} />
        </View>
    );
}

const styles = StyleSheet.create({
    card: {
        flexDirection: 'row',
        borderRadius: 8,
        backgroundColor: '#fff',
        marginBottom: 10,
        padding: 10,
        justifyContent: 'center',
        alignItems: 'center',
        shadowColor: '#000',
        shadowOffset: { width: 0, height: 2 },
        shadowOpacity: 0.3,
        shadowRadius: 3,
        elevation: 5,
    },
    textContainer: {
        flex: 1,
    },
    cardImage: {
        width: 80,
        height: 80,
        resizeMode: 'contain',
        borderRadius: 8,
    },
    cardName: {
        fontSize: 25,
        fontWeight: 'bold',
        color: '#000',
    },
    cardDescription: {
        fontSize: 18,
        color: '#000',
        marginTop: 4,
    },
    cardPrice: {
        fontSize: 18,
        color: '#000',
        marginTop: 4,
    },
});
