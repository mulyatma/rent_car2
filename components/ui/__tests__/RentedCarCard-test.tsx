import React from 'react';
import renderer from 'react-test-renderer';
import RentedCarCard from '../RentedCarCard';

describe('RentedCarCard Component', () => {
  it('renders correctly with given props', () => {
    const mockCar = {
      carId: {
        nameCar: 'Toyota Avanza',
        img: 'https://example.com/car.jpg',
      },
      startDate: '2023-01-01',
      endDate: '2023-01-07',
      totalPrice: 3000000,
    };

    const tree = renderer.create(<RentedCarCard car={mockCar} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('formats dates and price correctly', () => {
    const mockCar = {
      carId: {
        nameCar: 'Toyota Avanza',
        img: 'https://example.com/car.jpg',
      },
      startDate: '2023-01-01',
      endDate: '2023-01-07',
      totalPrice: 3000000,
    };

    const component = renderer.create(<RentedCarCard car={mockCar} />);
    const instance = component.root;

    // Check formatted dates
    const startDateText = instance.findByProps({ children: 'Sewa dari: 1/1/2023' });
    expect(startDateText).toBeTruthy();

    const endDateText = instance.findByProps({ children: 'Sampai: 7/1/2023' });
    expect(endDateText).toBeTruthy();

    // Check formatted price
    const priceText = instance.findByProps({ children: 'Total harga: Rp 3.000.000' });
    expect(priceText).toBeTruthy();
  });
});
