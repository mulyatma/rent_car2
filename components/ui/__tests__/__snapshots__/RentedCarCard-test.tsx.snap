// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RentedCarCard Component renders correctly with given props 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderRadius": 8,
      "elevation": 5,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 10,
      "padding": 10,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 3,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        {
          "color": "#000",
          "fontSize": 25,
          "fontWeight": "bold",
        }
      }
    >
      Toyota Avanza
    </Text>
    <Text
      style={
        {
          "color": "#000",
          "fontSize": 18,
          "marginTop": 4,
        }
      }
    >
      Sewa dari: 
      1/1/2023
    </Text>
    <Text
      style={
        {
          "color": "#000",
          "fontSize": 18,
          "marginTop": 4,
        }
      }
    >
      Sampai: 
      7/1/2023
    </Text>
    <Text
      style={
        {
          "color": "#000",
          "fontSize": 18,
          "marginTop": 4,
        }
      }
    >
      Total harga: Rp 
      3.000.000
    </Text>
  </View>
  <Image
    source={
      {
        "uri": "https://example.com/car.jpg",
      }
    }
    style={
      {
        "borderRadius": 8,
        "height": 80,
        "resizeMode": "contain",
        "width": 80,
      }
    }
  />
</View>
`;
